changelogs/unreleased/persistent_gc_state.md

```diff
@@ -0,0 +1,4 @@
+## feature/replication
```

Comment:
IMO reboostrap should ignore replica-local records and proceed anyway because by design replica-local records can't be used for storing information reliably.

Reply:
The question is what to do with it now, because it doesn't work. We file a new bug ticket to fix it, or fix it now, or remove this feature completely?

Reply:
There is a ticket: #5166. We just have to decide what to do with replica-local records. I think we can proceed with autorebootstrap ignoring the local records, but probably we need to put aside the node's existing xlog and snap files (rename them to `*.xlog.conflicted` or something like that), so that the user has a chance to recover his local data.

Reply:
IMHO, we should not fix this in the scope of this PR, there's a separate ticket for that, in scope of which we should write an RFC on how we want to fix this, since we cannot achieve consensus on this question. Ones say, that we can simply drop local spaces, other ones - local spaces are persistent and we cannot lose data without user intervention. I'm leaning towards the latter.

> I think we can proceed with autorebootstrap ignoring the local records, but we probably need to put aside the node's existing xlog and snap files (rename them to *.xlog.conflicted or something like that), so that the user has a chance to recover his local data after an unexpected rebootstrap.

Good solution, but there're issues with it, we need to address. The problem here, that we cannot know, that there's user local data on disk until we fully recover snap and xlogs. This one is not cheap. And if we decide to always rename files, when local component is > 0, then we'll never delete snap/xlogs after rebootstrap, even though we don't need `_gc_consumers` or raft to be preserved. It seems, we strongly missing vclock component, which we can always drop on rebootstap, all raft and system local spaces will be there.

The alternative is introducing a trigger, which will run on autorebootstrap. Users will decide, whether they want to:
  * preserve local data (and then they'll rename files with `fio`) and proceed;
  * silently proceed erasing all the local data
  * throw an error.
  
Since the majority of users doesn't care about local data, the default will be the second option (?). We'll give users a hand to preserve data, if they really want. But we all know, that such trigger won't be ever used, but our conscience is clear, since they have such opportunity.

Reply:
Ah, ok, if there is already a ticket, then it is fine. It just becomes hotter now, because since this PR the auto-rebootstrap doesn't work at all, while before it could at least work somehow sometimes.